src/Tasks.UnitTests/CodeTaskFactoryTests.cs

```diff
@@ -1200,6 +1203,83 @@ public void EmbedsGeneratedFileInBinlogWhenFailsToCompile()
             CodeTaskFactoryEmbeddedFileInBinlogTestHelper.BuildAndCheckForEmbeddedFileInBinlog(
                 FactoryType.CodeTaskFactory, "HelloTask", taskXml, false);
         }
+
+        [Fact]
+        public void ShouldEmitSingleGeneratedFileIntoBinlog()
+        {
+            using var env = TestEnvironment.Create();
+
+            // Define task XML for Import.targets
+            string taskXml = @"
+                <Project>
+                  <UsingTask
+                    TaskName=""CustomTask""
+                    TaskFactory=""CodeTaskFactory""
+                    AssemblyFile=""$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll"">
+                    <ParameterGroup>
+                      <InputParameter ParameterType=""System.String"" />
+                      <OutputParameter ParameterType=""System.String"" Output=""True"" />
+                    </ParameterGroup>
+                    <Task>
+                      <Using Namespace=""System"" />
+                      <Code Type=""Fragment"" Language=""cs"">
+                        <![CDATA[
+                            Console.WriteLine(this.InputParameter);
+                            this.OutputParameter = ""Hello "" + this.InputParameter;
+                        ]]>
+                      </Code>
+                    </Task>
+                  </UsingTask>
+                </Project>";
+
+            TransientTestFile importTargetsFile = env.CreateFile("Import.targets", taskXml);
+
+            // Define Another.proj content
+            string anotherContent = $@"<Project>
+                <Import Project=""{importTargetsFile.Path.Replace("\\", "/")}"" />
```

Comment:
No need to change the path delimiter if I remember correctly. Other places as well.